mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/query/parsing/nativequery/impl/NativeQueryParser.java

```diff
@@ -57,301 +67,300 @@
  * @author Guillaume Smet
  */
 @BuildParseTree
-public class NativeQueryParser extends BaseParser<MongoDBQueryDescriptorBuilder> {
-
-	final MongoDBQueryDescriptorBuilder builder;
+public class NativeQueryParser extends BaseParser<MongoDBQueryDescriptor> {
+
+	// TODO There are many more query types than what we support.
+	protected static List<String> RESERVED = Arrays.asList(
+			"findOne", "findAndModify", "find", "insertOne", "insertMany", "insert",
+			"remove", "deleteOne", "deleteMany", "updateOne", "updateMany", "update",
+			"replaceOne", "count", "aggregate", "distinct", "mapReduce" );
+	protected MongoDBQueryDescriptorBuilder builder = new MongoDBQueryDescriptorBuilder();
+	protected CliQueryInfo cliQueryInfo = new CliQueryInfo();
```

Comment:
So following what I said above about the cliQueryInfo, we should either move it to the query builder. Or have a stack with several values in it.